Update docs/monthly-meeting/2024-01.md (#97)
python/docs-community: docs/monthly-meeting/2024-01.md

> 60 second updates on things you have been up to, questions you have, or developments you think people should know about. Please add yourself, and if you do not have an update to share, you can pass.

> This is a place to make announcements (without a need for discussion). This is also a great place to give shout-outs to contributors! We'll read through these at the beginning of the meeting.

* The Steering Council has [accepted](https://discuss.python.org/t/pep-732-the-python-documentation-editorial-board/36710/9?u=hugovk) [PEP 732](https://peps.python.org/pep-0732/) ("The Python Documentation Editorial Board") 🎉

   - [Carol] First meeting will be January 8, 2024.
   - [Carol] Looking at docs and website from a new user's perspective
   - [Hugo] What would be the best way to contact the Board? [Guido] That's already on the agenda. [Carol] For the short term we could use the docs community repo too.

* I [Jim] appreciate how considerate this group is. The particular incident motivating my appreciation is that I got mentioned in the minutes of last meeting, and several people took pains to be sure that my name was presented there as I wanted it to be. Thank you.

> * [Ryan] We made the update to the TOML documentation we talked about on the last call and got my first Python pull request merged!

Reviewer: Indent size and consistency

## Discussion

> - [Petr] Inclusive language
> 
>   The `tkinter`, `pty`/`termios`, `sqlite3` (maybe more?) use `master/slave` terminology in API and docs. What's the best way to come up with, and switch to, better terms?
> 
>   Is there an expert* I should talk to?
> 
>   - [Carol] Thursday Bram has authored a Responsible Communications book which may have helpful guidelines. Other communities have used `leader/follower`. [Linux PR on the subject](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb)
>   - [Hugo] Some suggestions: https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave (Carol: I like the suggestions in this document for different usages.)
>   - [CAM] Earlier BPO issue: [python/cpython#78786](https://github.com/python/cpython/issues/78786)
> 
>   - To do
>       - Check and update the devguide for new docs not to use terms and suggest alternatives
>       - Determine process for future changes of existing terminology
>       - Consider adding a statement to the documentation about existing code
>   - Petr will bounce ideas off Daniele :)
> 
> 

Reviewer: ```suggestion
- [Petr] The `tkinter`, `pty`/`termios`, `sqlite3` (maybe more?) use `master/slave` terminology in API and docs. What's the best way to come up with, and switch to, better terms? Is there an expert I should talk to?
  - [Carol] Thursday Bram has authored a Responsible Communications book which may have helpful guidelines. Other communities have used `leader/follower`. [Linux PR on the subject](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb)
  - [Hugo] Some suggestions: https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave (Carol: I like the suggestions in this document for different usages.)
  - [CAM] Earlier BPO issue: [python/cpython#78786](https://github.com/python/cpython/issues/78786)
  - Next steps:
      - Check and update the devguide for new docs not to use terms and suggest alternatives
      - Determine process for future changes of existing terminology
      - Consider adding a statement to the documentation about existing code
  - Petr will bounce ideas off Daniele :)

```

> - Ryan tried introducing argument lists (like in [sqlite3](https://docs.python.org/3/library/sqlite3.html#sqlite3.connect)) to the collections module, didn't work too well as the functions take one or two args and they're pretty obvious. Petr recommended looking at `subprocess.Popen`.
>   - [Carol] Ryan, feel free to ping me when you have an example since I'm very familiar with the scientific Python docs.
>   - Guido doesn't like forcing people to stick to a template
>   - Daniele mentions there are two kinds of users, one of which -- a forensic analyst -- could need very strictly formatted documentation
>   - [Guido] The bulleted list makes it less likely to fit everything on a screen. Vertical space is a precious resource.
>   - [Carol] Doc usage has changed thanks to intellisense, nowadays you get lists of arguments on hovering.
>   - [Carol] Areas where users ask for help most should get most attention
>   - [Ezio] There is some value in a consistent format for return values and exceptions; sometimes it's hard to find them currently. Arguments are usually explained well in the description.
>   - [Daniele] It doesn't need to be machine-readable. It should always be maximally human-comfortable.
>   - [Jim] It sounds like the framing of this discussion is, source text targeting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?
>   - [Ned] We write docs twice -- once for the docstrings and once in `rst`. That's more work, but it means we can have different strategies for each use case. [Guido] Changing that would involve several PEPs.
>   - [CAM] Sphinx has ways to generate docs for a whole module, or just a single function, from docstrings. But yes, starting to use that would need a PEP.
>   - [CAM] The strategy we took for `sqlite3` was to focus on the functions with most arguments, where the bulleted lists have most benefit, and leave the smaller functions for later, when we have more experience with the approach.
>   - [Carol] And we should keep in mind that our perspectives are not necessarily the users' perspective. If we do make major changes we should think about users and we should vet the changes.
>   - Ryan started contributing by trying to get the contract of [`collections.Counter.most_common`](https://docs.python.org/3/library/collections.html#collections.Counter.most_common), and not finding it in the docstring nor in the documentation.
>   - Carol: If you're coming from the scientific world, where the `?` in IPython gives you the docstring, that is often the first thing you look at.
>   - Ryan will take a look at only documenting return type and "raises" in a structured way, leaving parameters in the prose.
>   - Guido warns that `collections.Counter` is not that typical.

Reviewer: ```suggestion
- Ryan tried introducing argument lists (like in [sqlite3](https://docs.python.org/3/library/sqlite3.html#sqlite3.connect)) to the collections module, didn't work too well as the functions take one or two arguments and they're pretty obvious. Petr recommended looking at `subprocess.Popen`.
  - [Carol] Ryan, feel free to ping me when you have an example since I'm very familiar with the scientific Python docs.
  - Guido doesn't like forcing people to stick to a template
  - Daniele mentions there are two kinds of users, one of which -- a forensic analyst -- could need very strictly formatted documentation
  - [Guido] The bulleted list makes it less likely to fit everything on a screen. Vertical space is a precious resource.
  - [Carol] Doc usage has changed thanks to intellisense, nowadays you get lists of arguments on hovering.
  - [Carol] Areas where users ask for help most should get most attention.
  - [Ezio] There is some value in a consistent format for return values and exceptions; sometimes it's hard to find them currently. Arguments are usually explained well in the description.
  - [Daniele] It doesn't need to be machine-readable. It should always be maximally human-comfortable.
  - [Jim] It sounds like the framing of this discussion is, source text targeting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?
  - [Ned] We write docs twice -- once for the docstrings and once in `rst`. That's more work, but it means we can have different strategies for each use case.
    - [Guido] Changing that would involve several PEPs.
  - [CAM] Sphinx has ways to generate docs for a whole module, or just a single function, from docstrings. But yes, starting to use that would need a PEP.
  - [CAM] The strategy we took for `sqlite3` was to focus on the functions with most arguments, where the bulleted lists have most benefit, and leave the smaller functions for later, when we have more experience with the approach.
  - [Carol] And we should keep in mind that our perspectives are not necessarily the users' perspective. If we do make major changes we should think about users and we should vet the changes.
  - Ryan started contributing by trying to get the contract of [`collections.Counter.most_common`](https://docs.python.org/3/library/collections.html#collections.Counter.most_common), and not finding it in the docstring nor in the documentation.
  - [Carol] If you're coming from the scientific world, where the `?` in IPython gives you the docstring, that is often the first thing you look at.
  - Ryan will take a look at only documenting return type and "raises" in a structured way, leaving parameters in the prose.
  - Guido warns that `collections.Counter` is not that typical.
```

- If we have time, would like to hear from Daniele on the [thread he started on the structure of the Python docs](https://discuss.python.org/t/diataxis-and-python-documentation/41836) as to the takeways from that so far and next steps there.

>   - Discussion between Daniele and Ezio on the Python tutorial

Reviewer: Should we add more context here @ezio-melotti @EvilDMP?

Reviewer 2: We could add the summary of the Discord thread where the discussion started.

Please take a second to read through it!

## Roll call

(Name / `@GitHubUsername` *[/ Discord, if different]*)
- Carol Willing / `@willingc` / carolwilling
- Petr Viktorin / `@encukou`
- Ezio Melotti / `@ezio-melotti`
- Hugo van Kemenade / `@hugovk`
- Guido van Rossum / `@gvanrossum`
- Daniele / `@EvilDMP` / danieleprocida
- Ege Akman / `@egeakman`
- Ned / `@nedbat`
- Ryan / `@ryan-duve`
- Jim / `@JDLH`
- CAM / `@CAM-Gerlach`
- Usman Akinyemi / `@Unique-Usman`
- Bradley / `@shenanigansd`

## Reports and celebrations

> 60 second updates on things you have been up to, questions you have, or developments you think people should know about. Please add yourself, and if you do not have an update to share, you can pass.

> This is a place to make announcements (without a need for discussion). This is also a great place to give shout-outs to contributors! We'll read through these at the beginning of the meeting.

* The Steering Council has [accepted](https://discuss.python.org/t/pep-732-the-python-documentation-editorial-board/36710/9?u=hugovk) [PEP 732](https://peps.python.org/pep-0732/) ("The Python Documentation Editorial Board") 🎉

>    - [willingc] First meeting will be January 8, 2024.
>    - [Carol] Looing at docs and website from a new user's perspective
>    - [Hugo] What would be the best way to contact the Board? [Guido] That's already on the agenda. [Carol] For the short term we could use the docs community repo too.

Reviewer 2: ```suggestion
   - [Carol] First meeting will be January 8, 2024.
   - [Carol] Looking at docs and website from a new user's perspective
   - [Hugo] What would be the best way to contact the Board? [Guido] That's already on the agenda. [Carol] For the short term we could use the docs community repo too.
```

* I [Jim] appreciate how considerate this group is. The particular incident motivating my appreciation is that I got mentioned in the minutes of last meeting, and several people took pains to be sure that my name was presented there as I wanted it to be. Thank you.

* [Ryan] We made the update to the TOML documentation we talked about on the last call and got my first Python pull request merged!

## Discussion

- [Petr] Inclusive language

  The `tkinter`, `pty`/`termios`, `sqlite3` (maybe more?) use `master/slave` terminology in API and docs. What's the best way to come up with, and switch to, better terms?

  Is there an expert* I should talk to?

>   - [willingc] Thursday Bram has authored a Responsible Communications book which may have helpful guidelines. Other communities have used `leader/follower`. [Linux PR on the subject](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb)
>   - [Hugo] Some suggestions: https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave (willingc: I like the suggestions in this document for different usages.)

Reviewer 2: ```suggestion
  - [Carol] Thursday Bram has authored a Responsible Communications book which may have helpful guidelines. Other communities have used `leader/follower`. [Linux PR on the subject](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb)
  - [Hugo] Some suggestions: https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave (Carol: I like the suggestions in this document for different usages.)
```

  - [CAM] Earlier BPO issue: [python/cpython#78786](https://github.com/python/cpython/issues/78786)

  - To do

>       - Check and update the dev guide for new docs not to use terms and suggest alternatives

Reviewer 2: ```suggestion
      - Check and update the devguide for new docs not to use terms and suggest alternatives
```

      - Determine process for future changes of existing terminology
      - Consider adding a statement to the documentation about existing code
  - Petr will bounce ideas off Daniele :)

- Ryan tried introducing argument lists (like in [sqlite3](https://docs.python.org/3/library/sqlite3.html#sqlite3.connect)) to the collections module, didn't work too well as the functions take one or two args and they're pretty obvious. Petr recommended looking at `subprocess.Popen`.

>   - [willingc] Ryan, feel free to ping me when you have an example since I'm very familiar with the scientific python docs.

Reviewer 2: ```suggestion
  - [Carol] Ryan, feel free to ping me when you have an example since I'm very familiar with the scientific Python docs.
```

  - Guido doesn't like forcing people to stick to a template
  - Daniele mentions there are two kinds of users, one of which -- a forensic analyst -- could need very strictly formatted documentation
  - [Guido] The bulleted list makes it less likely to fit everything on a screen. Vertical space is a precious resource.
  - [Carol] Doc usage has changed thanks to intellisense, nowadays you get lists of arguments on hovering.
  - [Carol] Areas where users ask for help most should get most attention
  - [Ezio] There is some value in a consistent format for return values and exceptions; sometimes it's hard to find them currently. Arguments are usually explained well in the description.
  - [Daniele] It doesn't need to be machine-readable. It should always be maximally human-comfortable.

>   - [Jim] It sounds like the framing of this discussion is, source text targetting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?

Reviewer 2: ```suggestion
  - [Jim] It sounds like the framing of this discussion is, source text targeting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?
```

- [Carol] always goes back to the users and what their pathways and needs are. The tutorial can't fit all users -- absolute beginners, people for whom Python is the first language, people coming from other languages. The current tutorial doesn't work for completely new users.
- [CAM] The current tutorial targets people who are already familiar with programming languages. Perhaps we should have separate tutorials for complete beginners and people coming from specific languages.
- [Daniele] One drawback of the current tutorial is lots of preamble saying why you want to learn Python. But the user is already here, wanting to learn Python. We should start directly with the examples. Also, we don't need to cover the edge cases right after people do something for the first time. Long explanations detract from the tutorial.
- [Ezio] It seems that for the tutorial, examples work better than prose, for both new and experienced users. Also, we could have cheatsheets for people coming from other languages.
- [Carol] There's a bigger pressing need for onboarding people from different groups, like people who don't know how to use the command line.

> ## Follow-ups from previous meeting(s)

Reviewer 3: We can remove this section, the bits we covered have been added above.

Reviewer 3: I meant remove this section from this line down, not the preceding paragraph.

Author: "## Follow-ups from previous meeting(s)" ?

Reviewer 3: Yes, starting from and including:

`## Follow-ups from previous meeting(s)`

Up to but excluding:

`## Next meeting`

And re-instate the deleted paragraph.

  - [Carol] Looking at docs and website from a new user's perspective
  - [Hugo] What would be the best way to contact the Board? [Guido] That's already on the agenda. [Carol] For the short term we could use the docs community repo too.

- I [Jim] appreciate how considerate this group is. The particular incident motivating my appreciation is that I got mentioned in the minutes of last meeting, and several people took pains to be sure that my name was presented there as I wanted it to be. Thank you.

- [Ryan] We made the update to the TOML documentation we talked about on the last call and got my first Python pull request merged!

## Discussion

- [Petr] The `tkinter`, `pty`/`termios`, `sqlite3` (maybe more?) use `master/slave` terminology in API and docs. What's the best way to come up with, and switch to, better terms? Is there an expert I should talk to?
  - [Carol] Thursday Bram has authored a Responsible Communications book which may have helpful guidelines. Other communities have used `leader/follower`. [Linux PR on the subject](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb)

>   - [Hugo] Some suggestions: https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave (Carol: I like the suggestions in this document for different usages.)

Reviewer 3: Please turn this into a click able link.

Author: which part of the text exactly ?

Reviewer 3: The URL:
`https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave`

Reviewer 2: Aren't links automatically clickable in MD?
See e.g. https://www.ietf.org/archive/id/draft-knodel-terminology-14.html#name-master-slave, with no special markup.

Reviewer 3: This one wasn't clickable when I originally checked the preview.

Reviewer 3: See also the hackmd link at the top.

  - Daniele mentions there are two kinds of users, one of which -- a forensic analyst -- could need very strictly formatted documentation
  - [Guido] The bulleted list makes it less likely to fit everything on a screen. Vertical space is a precious resource.
  - [Carol] Doc usage has changed thanks to intellisense, nowadays you get lists of arguments on hovering.
  - [Carol] Areas where users ask for help most should get most attention.
  - [Ezio] There is some value in a consistent format for return values and exceptions; sometimes it's hard to find them currently. Arguments are usually explained well in the description.
  - [Daniele] It doesn't need to be machine-readable. It should always be maximally human-comfortable.
  - [Jim] It sounds like the framing of this discussion is, source text targeting a big module documentation page with docs for all that module's methods. There are other possible targets: docstrings, tooltips text. Are there other targets for doc source which we should consider in this discussion?
  - [Ned] We write docs twice -- once for the docstrings and once in `rst`. That's more work, but it means we can have different strategies for each use case.

>     - [Guido] Changing that would involve several PEPs.

Reviewer 4: ```suggestion
    - [Guido] Changing that would involve several PEPs.
    - [Ned] Right, I wasn't suggesting we change it, just ensuring everyone knew.
```
